platforms/core-configuration/configuration-cache/build.gradle.kts

```diff
@@ -40,8 +40,8 @@ tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach
 dependencies {
     api(projects.concurrent)
     api(projects.javaLanguageExtensions)
-    api(projects.serialization)
     api(projects.serviceProvider)
+    api(projects.configurationProblemsBase)
     api(project(":base-services"))
     api(project(":build-operations"))
     // TODO - it might be good to allow projects to contribute state to save and restore, rather than have this project know about everything
```

Comment:
Ah, this refers to the design you mentioned we want to pursue, which would de-invert the dependencies.